src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -945,6 +945,8 @@ async Task CreateResourceExecutablesAsyncCore(IResource resource, IEnumerable<Ap
 
                 try
                 {
+                    await ProcessUrls(resource, cancellationToken).ConfigureAwait(false);
```

Comment:
Does moving this to DcpExecutor impact adding URLs to custom resources? Or is the name just misleading 😁

Reply:
The previous `OnResourceStarting` event that I moved `ProcessUrls` out of is only called by the DCP. I believe this only would have been called for DCP resources.

Do you intend to `ProcessUrls` to be called for non-DCP resources? I assume non-DCP resources should have `ResourceUrlAnnotation` added to them directly.

Reply:
Hmm this might actually be an oversight in the original implementation. We had intended for URLs to be supported on any resource type, not just resources with endpoints (which is why the `WithUrls` extension method is only constrained to `IResource`) but the code in `ApplicationOrchestrator` is only calling callbacks for `IResourceWithEndpoints` resources. I'll log an issue to track this.

Reply:
#9128